pkg/services/ngalert/api/api_ruler.go

```diff
@@ -380,6 +399,15 @@ func (srv RulerSrv) updateAlertRulesInGroup(c *contextmodel.ReqContext, groupKey
 		}
 		return ErrResp(http.StatusInternalServerError, err, "failed to update rule group")
 	}
+
+	if srv.featureManager.IsEnabled(c.Req.Context(), featuremgmt.FlagAlertingSimplifiedRouting) && notificationSettingsChanged {
+		// This isn't strictly necessary since the alertmanager config is periodically synced.
+		err := srv.amRefresher.RefreshConfig(c.Req.Context(), c.SignedInUser.GetOrgID())
+		if err != nil {
+			srv.log.Warn("Failed to refresh Alertmanager config for org after change in notification settings", "org", c.SignedInUser.GetOrgID(), "error", err)
+		}
+	}
```

Comment:
what if we do feature flag check in MAM in RefreshConfig?

Reply:
I think having it here makes it clear this refresh is happening only because of simplified routing while technically `RefreshConfig` could be used outside of simplified routing. Eventually the check will go away though, so either way is fine with me.

Reply:
The way I think about this is that regardless of the feature flag status, the AlertRule model still have the settings. Therefore, from the rule API perspective, it's not necessary to be aware of the flag, and logic "if there are changes in notification settings, notify whoever is on another end of the RefreshConfig". The other side, i.e. implementaion of AmRefresher, can whether ignore call and log a warn message or proceed with refresh (ideally, async).